lnschema_core/dev/sqlmodel.py

```diff
@@ -163,6 +168,20 @@ def validate_relationship_types(model, user_kwargs):
             )
             raise TypeError(error_message) from None
 
+    # auto-populate foreign key fields associated with any relationship passed by the user (avoid missing key error during pydantic validation)
+    all_sqm_relationships = {**standard_sqm_relationships, **custom_sqm_relationships}
+    all_user_rel_keys = all_sqm_relationships.keys() & user_kwargs.keys()
+    for relationship_name in all_user_rel_keys:
+        # get names of foreign key fields associate with the relationship
+        fk_columns = getattr(model.__class__, relationship_name).property.target.primary_key.columns._all_columns
+        fk_col_names = [column.name for column in fk_columns]
+        rel_field_name = relationship_name.replace("_", "")  # this is delicate and relies on standards for naming fields
+        fk_fields = [(col_name, f"{rel_field_name}_{col_name}") for col_name in fk_col_names]
```

Comment:
I don't fully understand why so much "post-processing" is needed including the flaky string replace after calling: `getattr(model.__class__, relationship_name).property.target.primary_key.columns._all_columns`

Is the complexity stemming from going from `sa.Column` to `sqm.Field` without there being a proper API for making that identification?

Reply:
There should be an API that directly gives you the column name corresponding to the relationship:
```
sqmrelationship.__sa_relationship__.foreign_key -> column/field
```

If there is a link table involved, there is *no* `foreign_key`, but instead, what `sqlalchemy.relationship` calls `secondary` table.
```
sqmrelationship.__sa_relationship__.secondary -> link table
```